.github/workflows/hlint.yaml

```diff
@@ -0,0 +1,22 @@
+name: hlint
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  hlint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: 'Set up HLint'
+      uses: haskell/actions/hlint-setup@v2
+      with:
+        version: '3.1.6'
```

Comment:
Can we set it to the version we currently use, `3.4.1`? Hopefully there won't be any unexpected warnings then, and if yes, we should investigate why.

Reply:
One difference betwen the way it's run by the haskell-ci workflow and this action is that haskell-ci detects `default-language:` and passes `-XHaskell2010` to hlint, whereas the hlint-run action doesn't permit passing any command-line args at all. Not sure what difference that makes in practice. Passing `XHaskell2010` triggered a bug in hlint in the past which is fixed now, and old haskell-ci used to not pass it at all historically, and we never really cared whether it does or doesn't except for the bug it triggered, so it'll probably be fine.

Reply:
> Can we set it to the version we currently use, `3.4.1`? Hopefully there won't be any unexpected warnings then, and if yes, we should investigate why.

Sure, this should be possible.

> One difference betwen the way it's run by the haskell-ci workflow and this action is that haskell-ci detects `default-language:` and passes `-XHaskell2010` to hlint, whereas the hlint-run action doesn't detect anything and permit passing any command-line args at all even if we needed to. Not sure what difference that makes in practice.

hlint already turns on a lot of language extensions by default; we would definitely notice if something went wrong here, as we'd have a lot of files that fail to parse.  Pretty safe to assume that this won't make any difference.